cmd/clusterctl/client/cluster/mover.go

```diff
@@ -647,18 +654,16 @@ func waitReadyForMove(ctx context.Context, proxy Proxy, nodes []*node, dryRun bo
 		}
 		key := client.ObjectKeyFromObject(obj)
 
-		blockLogged := false
-		if err := retryWithExponentialBackoff(ctx, backoff, func(ctx context.Context) error {
-			if err := c.Get(ctx, key, obj); err != nil {
+		log.Info(fmt.Sprintf("Move blocked by %s annotation, waiting for it to be removed", clusterctlv1.BlockMoveAnnotation))
+		if err := retryWithExponentialBackoff(ctx, waitForResourceMoveUnblockedBackoff, func(ctx context.Context) error {
+			if err := retryWithExponentialBackoff(ctx, getResourceBackoff, func(ctx context.Context) error {
+				return c.Get(ctx, key, obj)
+			}); err != nil {
 				return errors.Wrapf(err, "error getting %s/%s", obj.GroupVersionKind(), key)
 			}
 
 			if _, exists := obj.GetAnnotations()[clusterctlv1.BlockMoveAnnotation]; exists {
-				if !blockLogged {
```

Comment:
Why do we no longer need this check? Is this just extra code that was supposed to be removed in #9246?